plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml

```diff
@@ -3,33 +3,55 @@
-                <case regex="AddInteger">
-                    <header name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.AddInteger"/>
-                    <send description="Send request to the endpoint">
+            <switch description="Route based on the Operation" source="//ArithmaticOperation/Operation" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+                <case regex="Add">
+                    <payloadFactory description="Build backend payload" media-type="xml">
+                        <format>
+                            <tem:AddInteger xmlns:tem="http://tempuri.org">
+                                <tem:Arg1>$1</tem:Arg1>
+                                <tem:Arg2>$2</tem:Arg2>
+                            </tem:AddInteger>
+                        </format>
+                        <args>
+                            <arg evaluator="xml" expression="//ArithmaticOperation/Arg1"/>
+                            <arg evaluator="xml" expression="//ArithmaticOperation/Arg2"/>
+                        </args>
+                    </payloadFactory>
+                    <header description="Set SOAPAction header" name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.AddInteger"/>
+                    <call description="Send request to the endpoint">
                         <endpoint key="NumberAdditionEP"/>
-                    </send>
+                    </call>
                 </case>
-                <case regex="DivideInteger">
-                    <header name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.DivideInteger"/>
-                    <send>
+                <case regex="Divide">
+                    <payloadFactory description="Build backend payload" media-type="xml">
+                        <format>
+                            <tem:DivideInteger xmlns:tem="http://tempuri.org">
+                                <tem:Arg1>$1</tem:Arg1>
+                                <tem:Arg2>$2</tem:Arg2>
+                            </tem:DivideInteger>
+                        </format>
+                        <args>
+                            <arg evaluator="xml" expression="//ArithmaticOperation/Arg1"/>
+                            <arg evaluator="xml" expression="//ArithmaticOperation/Arg2"/>
+                        </args>
+                    </payloadFactory>
+                    <header description="Set SOAPAction header" name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.DivideInteger"/>
+                    <call>
                         <endpoint key="NumberDivisionEP"/>
-                    </send>
+                    </call>
                 </case>
                 <default>
                     <payloadFactory description="Build custom payload to send back" media-type="xml">
```

Comment:
Shall we say "create response payload for unsupported operation"

Reply:
Fixed.